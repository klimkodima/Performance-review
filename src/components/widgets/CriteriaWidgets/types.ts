export type TData = {
  title: string;
  statisticValue: number;
  isCorrespondence: boolean;
};

export type TDataList = TData[];
