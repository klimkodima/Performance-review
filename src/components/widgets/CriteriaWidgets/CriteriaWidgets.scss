.criteria-container {
  margin-top: 2% !important;
}

.criteria-items-wrapper {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.criteria-item-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
}

.criteria-item {
  height: 350px;
  max-width: 45% !important;
  margin-right: 5% !important;
  margin-bottom: 4% !important;
  position: relative;
  background-color: #f6f6f6;
  border: 2px #d7d7d7 solid;
  border-radius: 5px;
}

.criteria-item-title {
  font-size: 1.5em;
  font-weight: bold;
  color: #7e7e7e;
}

.criteria-item-statistic {
  font-size: 26px;
  font-weight: bold;
  color: #f54470;
  margin-top: 80px;
}

.criteria-item-smile {
  position: absolute;
  bottom: 30px;
  right: 20px;
}
