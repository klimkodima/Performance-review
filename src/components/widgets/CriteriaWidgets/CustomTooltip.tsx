import { FC } from 'react';
import { useTranslation } from 'react-i18next';
import SentimentSatisfiedAltIcon from '@mui/icons-material/SentimentSatisfiedAlt';
import SentimentSatisfiedIcon from '@mui/icons-material/SentimentSatisfied';
import SentimentVeryDissatisfiedIcon from '@mui/icons-material/SentimentVeryDissatisfied';

import styled from '@emotion/styled';
import Tooltip, { TooltipProps, tooltipClasses } from '@mui/material/Tooltip';

type TCorrespondence = {
  correspondenceText: string;
  correspondenceImg: any;
};

const smileStyle = {
  width: '50px',
  height: '50px',
  borderRadius: '50%'
};

const CustomWidthTooltip = styled(({ className, ...props }: TooltipProps) => (
  <Tooltip {...props} classes={{ popper: className }} />
))({
  [`& .${tooltipClasses.tooltip}`]: {
    border: '2px #797979 solid',
    backgroundColor: '#fff',
    color: '#030303',
    maxWidth: 170,
    textAlign: 'center'
  }
});

const CustomTooltip: FC<{ percent: number }> = ({ percent }) => {
  const { t } = useTranslation('common', { keyPrefix: 'criteria' });
  const correspondenceStatus: TCorrespondence[] = [
    {
      correspondenceText: t('goodCorrespondenceText'),
      correspondenceImg: (
        <SentimentSatisfiedAltIcon sx={{ ...smileStyle, bgcolor: 'green' }} />
      )
    },
    {
      correspondenceText: t('normalCorrespondenceText'),
      correspondenceImg: (
        <SentimentSatisfiedIcon sx={{ ...smileStyle, bgcolor: 'yellow' }} />
      )
    },
    {
      correspondenceText: t('badCorrespondenceText'),
      correspondenceImg: (
        <SentimentVeryDissatisfiedIcon sx={{ ...smileStyle, bgcolor: 'red' }} />
      )
    }
  ];

  const [goodCorrespondence, normalCorrespondence, badCorrespondence] =
    correspondenceStatus;
  let correspondence: TCorrespondence = {
    correspondenceText: '',
    correspondenceImg: ''
  };
  if (percent >= 80 && percent <= 100) {
    correspondence = goodCorrespondence;
  }
  if (percent >= 60 && percent <= 80) {
    correspondence = normalCorrespondence;
  }
  if (percent >= 0 && percent <= 59) {
    correspondence = badCorrespondence;
  }

  return (
    <CustomWidthTooltip
      title={correspondence.correspondenceText}
      data-testid='criteria-item-tooltip'
      leaveDelay={200}
    >
      {correspondence.correspondenceImg}
    </CustomWidthTooltip>
  );
};

export default CustomTooltip;
