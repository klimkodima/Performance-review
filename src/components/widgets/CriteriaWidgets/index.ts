export { default as CriteriaWidgets } from './CriteriaWidgets';
