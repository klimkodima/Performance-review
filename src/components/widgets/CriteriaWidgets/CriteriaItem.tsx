import { FC } from 'react';
import { useTranslation } from 'react-i18next';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';

import CustomTooltip from './CustomTooltip';
import { TData } from './types';

const CriteriaItem: FC<{ data: TData }> = ({ data }) => {
  const { title, statisticValue, isCorrespondence } = data;
  const { t } = useTranslation('common', { keyPrefix: 'criteria' });

  const getValueType = (titleName: string, value: number): string => {
    if (titleName === t('Average Performance')) {
      return value === 1 ? ` ${t('task/h')}` : ` ${t('tasks/h')}`;
    } else if (titleName === t('Support/Audits Hours')) {
      return t('h');
    } else return '%';
  };

  const valueType = getValueType(title, statisticValue);

  return (
    <Grid item xs={6} data-testid='criteria-item' className='criteria-item'>
      <Box className='criteria-item-wrapper'>
        <span className='criteria-item-title' data-testid='criteria-item-title'>
          {title}
        </span>
        <span
          className='criteria-item-statistic'
          data-testid='criteria-item-statistic'
        >
          {statisticValue}
          {valueType}
        </span>
        {isCorrespondence && (
          <div
            className='criteria-item-smile'
            data-testid='criteria-item-smile'
          >
            <CustomTooltip
              percent={statisticValue}
              data-testid='criteria-item-tooltip'
            />
          </div>
        )}
      </Box>
    </Grid>
  );
};

export default CriteriaItem;
