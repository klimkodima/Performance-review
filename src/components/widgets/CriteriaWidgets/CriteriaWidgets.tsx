import { FC } from 'react';
import { useTranslation } from 'react-i18next';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';

import CriteriaItem from './CriteriaItem';
import { TDataList, TData } from './types';

import './CriteriaWidgets.scss';

const CriteriaWidgets: FC = () => {
  const { t } = useTranslation('common', { keyPrefix: 'criteria' });
  const data: TDataList = [
    {
      title: t('Average Performance'),
      statisticValue: 1,
      isCorrespondence: false
    },
    {
      title: t('Support/Audits Hours'),
      statisticValue: 162,
      isCorrespondence: false
    },
    {
      title: t('Average Validity'),
      statisticValue: 96,
      isCorrespondence: false
    },
    {
      title: t('Hours Correspondence'),
      statisticValue: 90,
      isCorrespondence: true
    }
  ];
  return (
    <Grid
      item
      xs={8}
      className='criteria-container'
      data-testid='criteria-container'
    >
      <Box className='criteria-items-wrapper'>
        {data.map((item: TData) => (
          <CriteriaItem key={item.title} data={item} />
        ))}
      </Box>
    </Grid>
  );
};

export default CriteriaWidgets;
