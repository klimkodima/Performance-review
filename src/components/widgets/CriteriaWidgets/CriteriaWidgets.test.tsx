import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import CriteriaWidgets from './CriteriaWidgets';

describe('Criteria widget container tests', () => {
  test('Criteria widget snapshot', () => {
    const component = renderer.create(<CriteriaWidgets />).toJSON();
    expect(component).toMatchSnapshot();
  });
  test('Criteria widgets component rendered', () => {
    render(<CriteriaWidgets />);
    const container = screen.getByTestId('criteria-container');
    expect(container).toBeInTheDocument();
  });
  test('Criteria widgets component has classes', () => {
    render(<CriteriaWidgets />);
    const container = screen.getByTestId('criteria-container');
    expect(container).toHaveClass('criteria-container');
  });
  test('Criteria widgets component has 4 items', () => {
    render(<CriteriaWidgets />);
    const criteriaItems = screen.getAllByTestId('criteria-item');
    expect(criteriaItems.length).toBe(4);
  });
});

describe('Criteria widget item tests', () => {
  test('Criteria widgets item rendered', () => {
    render(<CriteriaWidgets />);
    const item = screen.getAllByTestId('criteria-item');
    expect(item[0]).toBeInTheDocument();
  });
  test('Criteria widgets item has classes', () => {
    render(<CriteriaWidgets />);
    const item = screen.getAllByTestId('criteria-item');
    expect(item[1]).toHaveClass('criteria-item');
  });
});

describe('Criteria widget item title tests', () => {
  test('Criteria widgets item title rendered', () => {
    render(<CriteriaWidgets />);
    const itemTitle = screen.getAllByTestId('criteria-item-title');
    expect(itemTitle[1]).toBeInTheDocument();
  });
  test('Criteria widgets item title has classes', () => {
    render(<CriteriaWidgets />);
    const itemTitle = screen.getAllByTestId('criteria-item-title');
    expect(itemTitle[1]).toHaveClass('criteria-item-title');
  });
  test('Criteria widgets item title has text', () => {
    render(<CriteriaWidgets />);
    const itemTitle = screen.getAllByTestId('criteria-item-title');
    expect(itemTitle[1]).toContainHTML('Support/Audits Hours');
  });
});

describe('Criteria widget item statistics tests', () => {
  test('Criteria widgets item statistic rendered', () => {
    render(<CriteriaWidgets />);
    const itemStatistics = screen.getAllByTestId('criteria-item-statistic');
    expect(itemStatistics[1]).toBeInTheDocument();
  });
  test('Criteria widgets item statistics has classes', () => {
    render(<CriteriaWidgets />);
    const itemStatistics = screen.getAllByTestId('criteria-item-statistic');
    expect(itemStatistics[1]).toHaveClass('criteria-item-statistic');
  });
  test('Criteria widgets item statistics has text', () => {
    render(<CriteriaWidgets />);
    const itemStatistics = screen.getAllByTestId('criteria-item-statistic');
    expect(itemStatistics[1]).toContainHTML('162');
  });
});

describe('Criteria widget item smile tests', () => {
  test('Criteria widgets item smile rendered', () => {
    render(<CriteriaWidgets />);
    const itemSmile = screen.getByTestId('criteria-item-smile');
    expect(itemSmile).toBeInTheDocument();
  });
  test('Criteria widgets item smile has classes', () => {
    render(<CriteriaWidgets />);
    const itemSmile = screen.getByTestId('criteria-item-smile');
    expect(itemSmile).toHaveClass('criteria-item-smile');
  });
  test('Criteria widgets item smile has tooltip', () => {
    render(<CriteriaWidgets />);
    const itemSmile = screen.getByTestId('criteria-item-smile');
    const itemTool = screen.getByTestId('criteria-item-tooltip');
    expect(itemSmile).toContainElement(itemTool);
  });
});
