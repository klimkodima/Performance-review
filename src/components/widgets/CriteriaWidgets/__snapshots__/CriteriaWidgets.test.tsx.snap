// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Criteria widget container tests Criteria widget snapshot 1`] = `
<div
  className="criteria-container"
  data-testid="criteria-container"
>
  <div
    className="criteria-item"
    data-testid="criteria-item"
  >
    <span
      className="criteria-item-title"
      data-testid="criteria-item-title"
    >
      Average Performance
    </span>
    <span
      className="criteria-item-statistic"
      data-testid="criteria-item-statistic"
    >
      5
       tasks/h
    </span>
  </div>
  <div
    className="criteria-item"
    data-testid="criteria-item"
  >
    <span
      className="criteria-item-title"
      data-testid="criteria-item-title"
    >
      Support/Audits Hours
    </span>
    <span
      className="criteria-item-statistic"
      data-testid="criteria-item-statistic"
    >
      162
      h
    </span>
  </div>
  <div
    className="criteria-item"
    data-testid="criteria-item"
  >
    <span
      className="criteria-item-title"
      data-testid="criteria-item-title"
    >
      Average Validity
    </span>
    <span
      className="criteria-item-statistic"
      data-testid="criteria-item-statistic"
    >
      96
      %
    </span>
  </div>
  <div
    className="criteria-item"
    data-testid="criteria-item"
  >
    <span
      className="criteria-item-title"
      data-testid="criteria-item-title"
    >
      Hours Correspondence
    </span>
    <span
      className="criteria-item-statistic"
      data-testid="criteria-item-statistic"
    >
      90
      %
    </span>
    <div
      className="criteria-item-smile"
      data-testid="criteria-item-smile"
    >
      <svg
        aria-hidden={true}
        aria-label="goodCorrespondenceText"
        aria-labelledby={null}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1icur68-MuiSvgIcon-root"
        data-mui-internal-clone-element={true}
        data-testid="criteria-item-tooltip"
        focusable="false"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        viewBox="0 0 24 24"
      >
        <circle
          cx="15.5"
          cy="9.5"
          r="1.5"
        />
        <circle
          cx="8.5"
          cy="9.5"
          r="1.5"
        />
        <circle
          cx="15.5"
          cy="9.5"
          r="1.5"
        />
        <circle
          cx="8.5"
          cy="9.5"
          r="1.5"
        />
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm0-2.5c2.33 0 4.32-1.45 5.12-3.5h-1.67c-.69 1.19-1.97 2-3.45 2s-2.75-.81-3.45-2H6.88c.8 2.05 2.79 3.5 5.12 3.5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
